acceptance/acceptance_test.go

```diff
@@ -229,1905 +99,1417 @@ func testWithoutSpecificBuilderRequirement(
-	subjectPack := func(name string, args ...string) *exec.Cmd {
-		return packCmd(packHome, packPath, name, args...)
-	}
-
 	it.Before(func() {
-		var err error
-		packHome, err = ioutil.TempDir("", "buildpack.pack.home.")
-		h.AssertNil(t, err)
+		pack = managers.NewPackExecutor(t, assetManager, managers.Current, registryConfig, assert)
 	})
 
 	it.After(func() {
-		h.AssertNil(t, os.RemoveAll(packHome))
+		pack.Cleanup()
 	})
 
 	when("invalid subcommand", func() {
 		it("prints usage", func() {
-			output, err := h.RunE(subjectPack("some-bad-command"))
-			h.AssertNotNil(t, err)
-			h.AssertContains(t, output, `unknown command "some-bad-command" for "pack"`)
-			h.AssertContains(t, output, `Run 'pack --help' for usage.`)
+			output, err := pack.RunWithCombinedOutput("some-bad-command")
+			assert.NotNil(err)
+
+			assert.Contains(output, `unknown command "some-bad-command" for "pack"`)
+			assert.Contains(output, `Run 'pack --help' for usage.`)
 		})
 	})
 
 	when("suggest-builders", func() {
 		it("displays suggested builders", func() {
-			output := h.Run(t, subjectPack("suggest-builders"))
-
-			h.AssertContains(t, output, "Suggested builders:")
-			h.AssertMatch(t, output, `Google:\s+'gcr.io/buildpacks/builder'`)
-			h.AssertMatch(t, output, `Heroku:\s+'heroku/buildpacks:18'`)
-			h.AssertMatch(t, output, `Paketo Buildpacks:\s+'gcr.io/paketo-buildpacks/builder:base'`)
-			h.AssertMatch(t, output, `Paketo Buildpacks:\s+'gcr.io/paketo-buildpacks/builder:full-cf'`)
-			h.AssertMatch(t, output, `Paketo Buildpacks:\s+'gcr.io/paketo-buildpacks/builder:tiny'`)
+			output := pack.SuccessfulRunWithOutput("suggest-builders")
+
+			assertOutput := assert.NewOutputAssertionManager(output)
+			assertOutput.IncludesSuggestedBuildersHeading()
+			assertOutput.IncludesGoogleBuilder()
+			assertOutput.IncludesHerokuBuilder()
+			assertOutput.IncludesPaketoBuilders()
 		})
 	})
 
 	when("suggest-stacks", func() {
 		it("displays suggested stacks", func() {
-			cmd := subjectPack("suggest-stacks")
-			output, err := cmd.CombinedOutput()
-			if err != nil {
-				t.Fatalf("suggest-stacks command failed: %s: %s", output, err)
-			}
-			h.AssertContains(t, string(output), "Stacks maintained by the community:")
+			output := pack.SuccessfulRunWithOutput("suggest-stacks")
+
+			assert.NewOutputAssertionManager(output).IncludesSuggestedStacksHeading()
 		})
 	})
 
 	when("set-default-builder", func() {
 		it("sets the default-stack-id in ~/.pack/config.toml", func() {
-			output := h.Run(t, subjectPack("set-default-builder", "gcr.io/paketo-buildpacks/builder:base"))
-			h.AssertContains(t, output, "Builder 'gcr.io/paketo-buildpacks/builder:base' is now the default builder")
+			builderName := "gcr.io/paketo-buildpacks/builder:base"
+
+			output := pack.SuccessfulRunWithOutput("set-default-builder", builderName)
+
+			assert.NewOutputAssertionManager(output).ReportsSettingDefaultBuilder(builderName)
```

Comment:
nit: I would add in an assertion that it exists in the `config.toml` as well, for completeness

Reply:
I intentionally haven't added any tests or functionality in this PR, my goal was to present the exact set of tests that exist on main. We should add this test, but probably as a different PR